packages/compass-sidebar/src/components/legacy/navigation-items.tsx

```diff
@@ -278,6 +280,22 @@ export function NavigationItems({
   } = useOpenWorkspace();
   const { hasWorkspacePlugin } = useWorkspacePlugins();
 
+  const [databasesFilterRegex, setDatabasesFilterRegex] =
+    useState<RegExp | null>(null);
+
+  const onDatabasesFilterChange = useCallback(
+    (filterRegex: RegExp | null) => setDatabasesFilterRegex(filterRegex),
+    [setDatabasesFilterRegex]
+  );
+
+  // When filtering, emit an event so that we can fetch all collections. This
+  // is required as a workaround for the synchronous nature of the current
+  // filtering feature
+  const appRegistry = useGlobalAppRegistry();
+  useEffect(() => {
+    appRegistry.emit('sidebar-filter-navigation-list');
+  }, [databasesFilterRegex, appRegistry]);
```

Comment:
Services should be only accessed through the store actions. I understand it that we are continuing to accumulate the tech debt with all this work, but please, let's add a todo to clean it up

Also please let's continue running this if filter is actually present, this operation is extremely expensive, even more so with multiple connections.

Reply:
gosh, thanks! this made me realise I missed to emit the 'sidebar-expand-database'. I didn't notice because this one was called immediately and fetched all collections!

Reply:
moved the two events emitting to thunk. feels a bit weird as the actions do not affect the state. but if we ever consolidate databases and connections in the store, at least we won't have to search for this bit.